### PR TITLE
2.0 docs overhaul

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Restructures the documentation for 2.x. 

@peri044: You need to implement the export function we discussed and make sure the docs match. Please just commit directly to this branch if there isnt another branch for it already. If there is let me know so I can adjust the docs. 

N.b. This adjusts some imports to make more sense from a user prespective (cc: @gs-olive)

Fixes: #1924  

## Type of change

Please delete options that are not relevant and/or add your own.

- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: docsrc/index.rst [hunk 1]
```diff
@@ -87,6 +116,7 @@ Python API Documenation
 * :ref:`torch_tensorrt_py`
 * :ref:`torch_tensorrt_logging_py`
 * :ref:`torch_tensorrt_ptq_py`
+* :ref:`torch_tensorrt_fx_dynamo`
```
Comment (gs-olive) [inline docsrc/index.rst:119]: When rendering, this link does not exist. Should it be just `torch_tensorrt_dynamo_py`?

File: docsrc/dynamo/dynamo_export.rst [hunk 2]
```diff
@@ -1,22 +1,22 @@
 .. _dynamo_export:
 
-Torch-TensorRT Dynamo Backend
+Torch-TensorRT Dynamo Frontend
```
Comment (gs-olive) [inline docsrc/dynamo/dynamo_export.rst:3]: Maybe this should be renamed "AOT Path". Otherwise, the section (currently) looks like:
```
Dynamo Frontend
|
--- Torch-TensorRT torch.compile Backend
--- Torch-TensorRT Dynamo Frontend
```
Additionally, the words "backend" and "frontend" are used often in both articles, so the real meaning becomes more challenging to discern.

File: docsrc/dynamo/torch_compile.rst
Comment (gs-olive) [inline docsrc/dynamo/torch_compile.rst:1]: In line with the above, could this be "Torch-TensorRT `torch.compile` JIT Path"?

File: docsrc/contributors/dynamo_converters.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    ) -> Union[tensorrt.ITensor, Sequence[tensorrt.ITensor]]:
+
+
+
+The decorator takes a number of arguments:
+
+    * key: Node target for which the converter is implemented for (for example, torch.ops.aten.leaky_relu.default)
+    * enabled: Whether the converter should be enabled as a converter that can be used in the converter registry
+    * capability_validator: A lambda that can take a ``torch.fx.Node`` and determine if the converter can properly handle this Node. If the validator returns ``False``, the subgraph partitioner will make sure this Node is run in PyTorch in the compiled graph.
+    * priority: Allows developers to override existing converters in the converter registry
```
Comment (gs-olive) [inline docsrc/contributors/dynamo_converters.rst:34]: Consider adding double backticks to each of the keyword arguments for clearer rendering

File: docsrc/contributors/dynamo_converters.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    * ctx : The current state of the compiler. Converters primarily will manipulate ctx.net which is the ``tensorrt.INetworkDefinition`` being constructed. Additional metadata including user provided settings is available in this struct as well.
+    * target: Target key in the ``call_module`` or ``call_function`` above. eg: ``torch.ops.aten_.leaky_relu.default``. Note that ``torch.ops.aten._leaky_relu`` is the ``OpOverloadPacket`` while ``torch.ops.aten_.leaky_relu.default`` is ``OpOverload``.
+    * args: The arguments being passed to a particular Node (as collected by the ``torch_tensorrt.dynamo.conversion.TRTInterpreter``). These arguments along with the kwargs are to be used to construct a specific TensorRT subgraph representing the current node in the INetworkDefinition.
+    * kwargs: The arguments being passed to a particular Node (as collected by the ``torch_tensorrt.dynamo.conversion.TRTInterpreter``).
+    * name: String containing the name of the target
```
Comment (gs-olive) [inline docsrc/contributors/dynamo_converters.rst:47]: Similarly to the above, add backticks to keyword arguments for formatting clarity

File: docsrc/contributors/dynamo_converters.rst [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    ) -> torch.Tensor:
+        return torch.add(
+            torch.mul(input_, beta), torch.mul(torch.matmul(mat1, mat2), alpha)
+        )
+
+You can modify the decompositions run by editing ``torch_tensorrt.dynamo.lowering.torch_enabled_decompositions`` and ``torch_tensorrt.dynamo.lowering.torch_disabled_decompositions``
+
+    Note: ``torch_tensorrt.dynamo.lowering.torch_enabled_decompositions`` and ``torch_tensorrt.dynamo.lowering.torch_disabled_decompositions`` must be disjoint sets and that the decompositions already defined in ``torch_tensorrt.dynamo.lowering`` will take precedence over torch lowering ops.
+
+Much of the time, this is significantly easier than implmenting a converter. So where possible, this is what should be tried first.
```
Comment (gs-olive) [inline docsrc/contributors/dynamo_converters.rst:131]: Minor typo `implmenting` --> `implementing`

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1]: It seems that `mypy`'s behavior on this file is strange - if the file containing the definition of `dynamo_tensorrt_converter` is also part of the PR, the linter will not complain about decorator typing. Otherwise, it requires `# type: ignore[misc]` on every decorator to pass. Is there a way we could exempt this file from that specific `mypy` check or upgrade the `mypy` version to see if this issue persists?

**Review by gs-olive — COMMENTED** [review]
Changes overall look great - added a few comments on documentation and linting.

File: docsrc/dynamo/torch_compile.rst
Comment (narendasan) [reply docsrc/dynamo/torch_compile.rst:1]: I think it makes sense to call it

* "TensorRT backend for `torch.compile`"
* "Torch-TensorRT Dynamo Frontend" or even "Compiling torch.export.ExportedPrograms"

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py
Comment (narendasan) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1]: Yeah lets look at it post release. It seems like this is most correct though
